test/fluree/db/ledger/api/downloaded.clj

```diff
@@ -170,27 +162,53 @@
+          status        (:status multi-res)
+          body          (-> multi-res :body bs/to-string json/parse)
+          collections   (into #{} (map #(:_collection/name %) (:coll body)))
+          predicates    (into #{} (map #(:_predicate/name %) (:pred body)))]
 
       (is (= 200 status))
 
-      ; The keys in the response are -> :request-time :aleph/keep-alive? :headers :status :connection-time :body
-      (is (not (empty? (remove nil? (map #(#{:request-time :aleph/keep-alive? :headers :status :connection-time :body} %)
-                                         responseKeys)))))
+      ; The keys in the response are -> :opts :body :headers :status
+      (is (test/contains-many? multi-res :opts :body :headers :status))
 
       ; Are all the predicates what we expect?
       (is (= collections #{"_rule" "nestedComponent" "_fn" "_predicate" "_setting" "chat" "_auth" "_user" "person" "_shard" "_tag" "comment" "_role" "_collection"}))
 
       ; Are some of the predicates we expect returned?
       (is (every? boolean (map #(predicates %) ["comment/nestedComponent" "person/stringUnique"]))))))
 
+(deftest sign-multi-query
+  (testing "sign-multi-query where collections are not named in alphanumeric order"
+    (let [private-key   (slurp "default-private-key.txt")
+          qry-str       "{\"collections\":{\"select\":[\"*\"],\"from\":\"_collection\"},\n       \"predicates\":{\"select\":[\"*\"],\"from\":\"_predicate\"},\n       \"_setting\":{\"select\":[\"*\"],\"from\":\"_setting\"},\n       \"_rule\":{\"select\":[\"*\"],\"from\":\"_rule\"},\n       \"_role\":{\"select\":[\"*\"],\"from\":\"_role\"},\n       \"_user\":{\"select\":[\"*\"],\"from\":\"_user\"}\n      }"
+          request       {:headers {"content-type" "application/json"}
+                         :body    qry-str}
+          q-endpoint    (str endpoint-url "multi-query")
+          signed-req    (http-signatures/sign-request :post q-endpoint request private-key)
+          resp          @(http/post q-endpoint signed-req)
+          status        (:status resp)
+          body          (some-> resp :body bs/to-string json/parse)
+          collections   (into #{} (map #(:_collection/name %) (:collections body)))
+          predicates    (into #{} (map #(:_predicate/name %) (:predicates body)))
+          roles         (into #{} (map #(:_role/id %) (:_role body)))]
+
+      (is (= 200 status))
+
+      ; The keys in the response are -> :opts :body :headers :status
+      (is (test/contains-many? resp :opts :body :headers :status))
+
+      ; Are all the predicates what we expect?
+      (is (= collections #{"_rule" "nestedComponent" "_fn" "_predicate" "_setting" "chat" "_auth" "_user" "person" "_shard" "_tag" "comment" "_role" "_collection"}))
+
+      ; Are some of the predicates we expect returned?
+      (is (every? boolean (map #(predicates %) ["comment/nestedComponent" "person/stringUnique"])))
```

Comment:
I think this can be simplified to:

`(is (every? predicates ["comment/nestedComponent" "person/stringUnique"]))` since any missing result `nil` should cause `every?` to return `false`.

Or you could just use `contains-many?` here too since all of the provided keys should be in the set (which also support `contains?` on values so you can pass the set as the `m` arg to `contains-many?` and the vec of string literals as `ks`).

Reply:
✅